cpp/src/quantiles/quantiles.cu

```diff
@@ -100,44 +63,19 @@ namespace{ // anonymous
       thrust::sort(rmm::exec_policy(stream)->on(stream), devarr, devarr+n);
     }
 
-    QuantileIndex qi(n, quant);
+    CUDF_EXPECTS(interpolation < cudf::N_QUANT_METHODS &&
+                 interpolation >= 0,
+      "Invalid quantile interpolation method");
 
-    switch( interpolation )
-    {
-    case GDF_QUANT_LINEAR:
-      singleMemcpy(hvalue[0], devarr+qi.lower_bound, stream);
-      singleMemcpy(hvalue[1], devarr+qi.upper_bound, stream);
-      cudf::interpolate::linear(result, hvalue[0], hvalue[1], qi.fraction);
-      break;
-    case GDF_QUANT_MIDPOINT:
-      singleMemcpy(hvalue[0], devarr+qi.lower_bound, stream);
-      singleMemcpy(hvalue[1], devarr+qi.upper_bound, stream);
-      cudf::interpolate::midpoint(result, hvalue[0], hvalue[1]);
-      break;
-    case GDF_QUANT_LOWER:
-      singleMemcpy(hvalue[0], devarr+qi.lower_bound, stream);
-      result = static_cast<RetT>( hvalue[0] );
-      break;
-    case GDF_QUANT_HIGHER:
-      singleMemcpy(hvalue[0], devarr+qi.upper_bound, stream);
-      result = static_cast<RetT>( hvalue[0] );
-      break;
-    case GDF_QUANT_NEAREST:
-      singleMemcpy(hvalue[0], devarr+qi.nearest, stream);
-      result = static_cast<RetT>( hvalue[0] );
-      break;
-
-    default:
-      return GDF_UNSUPPORTED_METHOD;
-    }
+    result = cudf::detail::select_quantile(devarr, n, quant, interpolation);
     
     return GDF_SUCCESS;
   }
 
   template<typename ColType,
            typename RetT = double>
   gdf_error trampoline_exact(gdf_column*  col_in,

```

Comment:
What's with "trampoline" here? I've always wondered if this naming is helpful at all. The inside of this function could just be moved into the functor and the functor renamed to `quantile_exact_functor`.